src/logging.cc

```diff
@@ -739,43 +743,61 @@ inline void LogDestination::SetEmailLogging(LogSeverity min_severity,
   LogDestination::addresses_ = addresses;
 }
 
-static void ColoredWriteToStderr(LogSeverity severity,
-                                 const char* message, size_t len) {
-  const GLogColor color =
-      (LogDestination::terminal_supports_color() && FLAGS_colorlogtostderr) ?
-      SeverityToColor(severity) : COLOR_DEFAULT;
+static void ColoredWriteToStderrOrStdout(bool is_stdout, LogSeverity severity,
+                                         const char* message, size_t len) {
+  // We also need to send logs to the stderr when the severity is
+  // higher or equal to the stderr threshold.
+  is_stdout = is_stdout && severity < FLAGS_stderrthreshold;
```

Comment:
I think the decision whether the output is written `stdout` or `stderr` should be made outside of this function. You can introduce a helper function if you need to.